### PR TITLE
[Keyboard] Adding YMDK Wings Keyboard

### PR DESCRIPTION
**Pull request by alittlepeace**
## Description

YMDK Wings is an Arisu/Alice-style keyboard with a nav cluster that started being sold at the end of 2020. Their website instructs people to use their own tool off their own fork of QMK, so I decided to create the files for use with the configurator. I've also created a VIA .json. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/wings/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/ymdk/wings/rules.mk:21]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Also missing a readme.md.

File: keyboards/ymdk/wings/rules.mk (hunk 1 above)
Comment (alittlepeace) [reply keyboards/ymdk/wings/rules.mk]: Fixed rules.mk and added readme

File: keyboards/ymdk/wings/info.json [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
+    "keyboard_name": "YMDK Wings",
+    "url": "http://ymdkey.com/",
+    "maintainer": "qmk",
+    "width": 20,
+    "height": 5.25,
+    "layouts": {
+        "LAYOUT_default": {
```
Comment (zvecr) [inline keyboards/ymdk/wings/info.json:8]:
```suggestion
        "LAYOUT": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ymdk/wings/info.json (hunk 2 above)
Comment (alittlepeace) [reply keyboards/ymdk/wings/info.json]: changed

File: keyboards/ymdk/wings/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/ymdk/wings/keymaps/via/keymap.c:18]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ymdk/wings/keymaps/via/keymap.c (hunk 3 above)
Comment (alittlepeace) [reply keyboards/ymdk/wings/keymaps/via/keymap.c]: fixed

**Review by fauxpark — APPROVED** [review]

File: keyboards/ymdk/wings/info.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
+{
+    "keyboard_name": "YMDK Wings",
+    "url": "http://ymdkey.com/",
+    "maintainer": "qmk",
+    "width": 20,
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/info.json:5]:
```suggestion
    "width": 20.25,
```

File: keyboards/ymdk/wings/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# YMDK Wings
+
+![Wings](https://cdn.shopify.com/s/files/1/0006/5227/9869/products/IMG_8044_2048x2048.jpg)
+
+A keyboard inspired by the Arisu, which is inspired by Lyn's EM7 and TGR Alice
+
+* Keyboard Maintainer: alittlepeace
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/readme.md:7]:
```suggestion
* Keyboard Maintainer: [alittlepeace](https://github.com/alittlepeace)
```

File: keyboards/ymdk/wings/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+* Hardware Supported: YMDK Wings
+* Hardware Availability: [YMDK](https://ymdkey.com/products/ymdk-wings-aluminum-cnc-top-bottom-qmk-pcb-full-assembly-keyboard-jade-navy-cream-switches)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ymdk/wings:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/readme.md:15]:
```suggestion
Flashing example for this keyboard:

    make ymdk/wings:default:flash

To reset the board into bootloader mode, hold the key at the top left of the keyboard while connecting the USB cable (also erases persistent settings).

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/ymdk/wings/wings.h [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "quantum.h"
+#pragma once
+
+
+#define XXX KC_NO
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/wings.h:21]: The PCB, according to the YMDK site, supports a 2u Backspace, even if the case doesn't.

```suggestion
#define LAYOUT_all( \
```

File: keyboards/ymdk/wings/info.json [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    "height": 5.25,
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/info.json:8]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/ymdk/wings/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/keymaps/default/keymap.c:19]:
```suggestion
  [0] = LAYOUT_all(
```

File: keyboards/ymdk/wings/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS,  KC_GRV,     KC_PGUP,  
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,                KC_PGDN,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                           MO(1),
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                    KC_UP,
+    KC_LCTL,          KC_LALT,          KC_SPC,  KC_LGUI,          KC_SPC,           KC_RALT,                           KC_LEFT, KC_DOWN, KC_RGHT 
+  ),
+
+  [1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/keymaps/default/keymap.c:27]:
```suggestion
  [1] = LAYOUT_all(
```

File: keyboards/ymdk/wings/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/keymaps/via/keymap.c:20]:
```suggestion
  [0] = LAYOUT_all(
```

File: keyboards/ymdk/wings/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS,  KC_GRV, KC_PGUP,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,          KC_PGDN,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    MO(1),
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,
+    KC_LCTL, KC_LALT,          KC_SPC,  KC_LGUI,          KC_SPC,           KC_RALT,                   KC_LEFT, KC_DOWN, KC_RGHT
+  ),
+
+  [1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/keymaps/via/keymap.c:28]:
```suggestion
  [1] = LAYOUT_all(
```

File: keyboards/ymdk/wings/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    _______,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,   KC_F11,  KC_F12,  _______, _______, _______,
+    _______, _______, _______, _______, RESET,   _______, _______, _______, _______, _______, _______, _______, _______, KC_DEL,           _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______,          _______,          _______, _______,          _______,          _______,                   _______, _______, _______
+  ),
+
+  [2] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/keymaps/via/keymap.c:36]:
```suggestion
  [2] = LAYOUT_all(
```

File: keyboards/ymdk/wings/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______,          _______,          _______, _______,          _______,          _______,                   _______, _______, _______
+  ),
+  [3] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/ymdk/wings/keymaps/via/keymap.c:43]:
```suggestion
  [3] = LAYOUT_all(
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/11693/files).

**Review by drashna — APPROVED** [review]

**Pull request merged**